pkg/plugin/framework/item_snapshotter_client.go

```diff
@@ -0,0 +1,213 @@
+		Plugin: recv.plugin,
+	}
+
+	res, err := recv.grpcClient.AppliesTo(context.Background(), req)
+	if err != nil {
+		return velero.ResourceSelector{}, fromGRPCError(err)
+	}
+
+	if res.ResourceSelector == nil {
+		return velero.ResourceSelector{}, nil
+	}
+
+	return velero.ResourceSelector{
+		IncludedNamespaces: res.ResourceSelector.IncludedNamespaces,
+		ExcludedNamespaces: res.ResourceSelector.ExcludedNamespaces,
+		IncludedResources:  res.ResourceSelector.IncludedResources,
+		ExcludedResources:  res.ResourceSelector.ExcludedResources,
+		LabelSelector:      res.ResourceSelector.Selector,
+	}, nil
+}
+
+func (recv ItemSnapshotterGRPCClient) AlsoHandles(input *isv1.AlsoHandlesInput) ([]velero.ResourceIdentifier, error) {
+	panic("implement me")
+}
+
+func (recv ItemSnapshotterGRPCClient) SnapshotItem(ctx context.Context, input *isv1.SnapshotItemInput) (*isv1.SnapshotItemOutput, error) {
+	itemJSON, err := json.Marshal(input.Item.UnstructuredContent())
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	backupJSON, err := json.Marshal(input.Backup)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	req := &proto.SnapshotItemRequest{
+		Plugin: recv.plugin,
+		Item:   itemJSON,
+		Backup: backupJSON,
+	}
+	res, err := recv.grpcClient.SnapshotItem(ctx, req)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	var updatedItem unstructured.Unstructured
+	if err := json.Unmarshal(res.Item, &updatedItem); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	additionalItems := unpackResourceIdentifiers(res.AdditionalItems)
+	handledItems := unpackResourceIdentifiers(res.HandledItems)
+
+	sio := isv1.SnapshotItemOutput{
+		UpdatedItem:      &updatedItem,
+		SnapshotID:       res.SnapshotID,
+		SnapshotMetadata: nil,
+		AdditionalItems:  additionalItems,
+		HandledItems:     handledItems,
+	}
+	return &sio, nil
+}
+
+func (recv ItemSnapshotterGRPCClient) Progress(input *isv1.ProgressInput) (*isv1.ProgressOutput, error) {
+	backupJSON, err := json.Marshal(input.Backup)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	req := &proto.ProgressRequest{
+		Plugin:     recv.plugin,
+		ItemID:     resourceIdentifierToProto(input.ItemID),
+		SnapshotID: input.SnapshotID,
+		Backup:     backupJSON,
+	}
+
+	res, err := recv.grpcClient.Progress(context.Background(), req)
+
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	up := isv1.ProgressOutput{
+		Completed:       res.Completed,
+		Err:             nil,
+		ItemsCompleted:  res.ItemsCompleted,
+		ItemsToComplete: res.ItemsToComplete,
+		Started:         time.Unix(res.Started, res.StartedNano),
+		Updated:         time.Unix(res.Updated, res.UpdatedNano),
+	}
+	return &up, nil
+}
+
+func (recv ItemSnapshotterGRPCClient) DeleteSnapshot(ctx context.Context, input *isv1.DeleteSnapshotInput) error {
+	req := &proto.DeleteItemSnapshotRequest{
+		Plugin:     recv.plugin,
+		Params:     input.Params,
+		SnapshotID: input.SnapshotID,
```

Comment:
Should `SnapshotMetadata` be included in this request?

Reply:
Yeah, I forgot to marshal/unmarshal those, thanks

Reply:
Created #4356 to follow up.